src/cargo/util/toml/mod.rs

```diff
@@ -2231,8 +2231,15 @@ fn verify_lints(
     for (tool, lints) in lints {
         let supported = ["cargo", "clippy", "rust", "rustdoc"];
         if !supported.contains(&tool.as_str()) {
-            let supported = supported.join(", ");
-            anyhow::bail!("unsupported `{tool}` in `[lints]`, must be one of {supported}")
+            let message = format!(
+                "unrecognized lint tool `lints.{tool}`
+
+specifying an unrecognized lint tool could lead to future breakage, as `cargo`
+reserves all of the `lints` namespace for itself and provides no compatibility
+guarantees when encountering unrecognized tools"
```

Comment:
I also considered:
```
`cargo` reserves all of the `lints` namespace for itself and provides no
future compatibility guarantees when encountering unrecognized tools, as they
may be supported in the future.
```

Reply:
I prefer this one.

Should we also suggest using `package.metadata`?

Reply:
I feel like this makes it sound like the lints table is special and this doesn't apply elsewhere.

Something to keep in mind is people might also be here from a typo.

Reply:
I think this should be the top level message:
```
unrecognized lint tool `{}`, must be one of {supported}
```

What about:
```
no compatibility guarantees are provided when specifying unrecognized manifest
key(s), as the entire namespace is reserved for future use. Tools that wish to
store package configuration should use `package.metadata`
```
or 
```
no compatibility guarantees are provided when specifying unrecognized manifest
key(s) as `cargo` reserves the entire namespace for future use. `package.metadata`
should be used for tools which would like to store package configuration
```

Reply:
I'm good with the now updated wording